File: frontend/src/app/shop/models/order-dto.ts
import { OrderItemDto } from "./order-item-dto";

export interface OrderDto {
    id?: string,
    userId?: string,
    shippingAddress?: string,
    shippingCity?: string,
    shippingPostalCode?: string,
    receiverName?: string,
    receiverPhone?: string,
    additionalNote?: string,
    totalAmount?: number,
    orderStatus?: string,
    orderItems?: OrderItemDto[]
}

File: backend/src/main/java/org/stevenguyendev/pcshopwebsite/dto/OrderDTO.java
package org.stevenguyendev.pcshopwebsite.dto;

import org.stevenguyendev.pcshopwebsite.model.OrderStatus;

import java.math.BigDecimal;
import java.util.Set;
import java.util.UUID;

public record OrderDTO(
        UUID id,
        UUID userId,
        String shippingAddress,
        String shippingCity,
        String shippingPostalCode,
        String receiverName,
        String receiverPhone,
        String additionalNote,
        BigDecimal totalAmount,
        OrderStatus orderStatus,
        String createdAt,
        String updatedAt,
        String createdBy,
        String updatedBy,
        Set<OrderItemDTO> orderItems
) {
}
